### ISSUE TITLE
Delete fork button on closed pull requests

### ISSUE THREAD

**Issue by michaelmior** [isaacs/github#673]
Sometimes I want to make a small single PR to a project to fix a minor change. GH makes it easy to make the change and send the PR. However, if I no longer want to keep my copy of the fork of the repository I have to manually navigate back to the fork find the settings and select delete. It would be great if there was also a "Delete fork" button.

**Comment by gnprice** [comment]
GitHub seems to have now added this feature!

![image](https://user-images.githubusercontent.com/28173/52829626-659a3f00-3082-11e9-8445-dbfe63740221.png)

... Unfortunately I'm seeing it on a repository I work in routinely, in which I have a bunch of other branches. In fact, I'm seeing it even though I have other branches *for other PRs* that I just sent yesterday and are still open. So if this button actually does what it says, that'll be a pretty destructive operation.

A better rule would be: show the "Delete fork" button when my fork *only* has branches that are contained in the upstream repo. (I.e., when `git rev-list --remotes=me --not --remotes=origin` would be empty.) In other words: show the button only if there isn't other work, unrelated to this PR, that it would destroy.

**Comment by joshuapoq** [comment]
I tried the button, for science, and it asks you to type the name of your fork before deletion. I think this button should be more destructively coloured. As this was a massive worry point for myself and the rest of the devs in my company.

**Comment by jskeet** [comment]
> In other words: show the button only if there isn't other work, unrelated to this PR, that it would destroy.

Potentially with some heuristic around other uses of the fork. Even if there are no other branches related to PRs, if GitHub can "tell" that my regular work process is to create PRs from a fork, and I've done so hundreds of times over the last few years, why would it think that right now I need an easy way of deleting the fork?

**Comment by markusicu** [comment]
I am also seeing this button now for a project where I am active. Very scary.
This kind of thing should be well-hidden, behind barbed wire, with dire warnings.
Maybe with the exception of creating a fork, making one branch, and deleting that one (usually after a pull request got merged or rejected), nothing else.

**Comment by brianjmiller** [comment]
Seeing it too, share same sentiment about its destructiveness. Also the wording is entirely misleading, it says "If you wish, you can delete your fork of **repo name**." except in the case I'm looking at (merging someone else's PR) it would delete *their* fork.

**Comment by micky2be** [comment]
As an admin on private repos I can see the **Delete fork** button on every PRs from anyone.
This scares me a bit. That one could destroy the work of others so easily.

**Comment by TPS** [comment]
The OP's suggestion is now available as a dropdown from a "Delete Branch" option that shows, so I'm thinking of closing this issue.  Further improvements could be noted here, but probably should be a new issue (crosslinked here).

What d'y'all think?

**Comment by markusicu** [comment]
> The OP's suggestion is now available as a dropdown from a "Delete Branch" option that shows, so I'm thinking of closing this issue.

What? No -- it got worse, not better: Now we have "Delete fork" in *two* places, once in the dropdown under "Delete branch", and when you do delete the branch again prominently advertised with its own button as before.

Please do not offer "Delete fork" on the PR page at all!

**Comment by rgommers** [comment]
> Please do not offer "Delete fork" on the PR page at all!

Indeed, or at the very least offer an option to turn it off via Settings. I get this button daily and *never* want to use it - one of these days I'll hit it by accident....

**Comment by alina-diana-muresan** [comment]
> Please do not offer "Delete fork" on the PR page at all!

I agree. I'm working on several forks on a constant basis, and I definitely do **not** want to delete the forks at any point from the PR page. While I can understand that others have need of this feature, there should absolutely be an option in the github settings to not show this on PRs.

Furthermore, one should not be able to delete someone else's forks like this. It's incredibly destructive.

**Comment by xmo-odoo** [comment]
>  In fact, I'm seeing it even though I have other branches for other PRs that I just sent yesterday and are still open.

Confirming that this is apparently still the case and is incredibly odd and dangerous.

**Comment by jaredj** [comment]
+1

**Comment by Cryptophobia** [comment]
I saw this button today and I peed a little in my pants. I'm a cat after all and litter-box trained, but this button still managed to scare me.

+0.5

**Comment by peter-dolkens** [comment]
This should be a setting, just like auto-delete branches is an option.

We've got devs who haven't yet fully transitioned to the "auto-delete branches" world, so are conditioned to manually clean up behind themselves - one day after a big crunch that's going to include cleaning up an important fork, and you know it's going to happen because they're already exhausted - just what they need.

**Comment by arnabanimesh** [comment]
This option should be provided during the creation of a pull request. A confirmation popup can be shown upon selecting the checkbox.

**Comment by JLLeitschuh** [comment]
I'm debating writing a GitHub action to do this for me automatically.

I also wish that this was easier to do across the board. I wrote a bot that generated 1,596 PRs which required just as many forks to be generated. I'd really like if this could be pre-configured against my account such that as these PRs were merged, they could auto-delete themselves, instead of requiring me to go to each one individually to delete them.
